client/dashboard/emails/hooks/use-domains.ts

```diff
@@ -2,9 +2,19 @@ import { SiteDomain } from '@automattic/api-core';
 import { siteDomainsQuery, sitesQuery } from '@automattic/api-queries';
 import { useQueries, useQuery } from '@tanstack/react-query';
 
+const isJetpackSlug = ( slug: string ): boolean => String( slug ).startsWith( 'jetpack_' );
+
 export const useDomains = (): { domains: SiteDomain[]; isLoading: boolean } => {
 	const { data: allSites, isLoading: isLoadingSites } = useQuery( sitesQuery() );
-	const sites = ( allSites ?? [] ).filter( ( site ) => site.capabilities.manage_options );
+	const sites = ( allSites ?? [] )
+		.filter( ( site ) => {
+			const product = site?.plan?.product_slug ?? null;
+			if ( product === null ) {
+				return true;
+			}
+			return ! isJetpackSlug( product );
```

Comment:
Nothing prevents us from creating new Jetpack plans in the future whose product slug does not start with `jetpack_`. Wouldn't be safer to check `site.jetpack && site.is_wpcom_atomic`?  Maybe even worth adding a new `isJetpack` util to https://github.com/Automattic/wp-calypso/blob/trunk/client/dashboard/utils/site-types.ts

Reply:
I think I could use the `isSelfHostedJetpackConnected` function in `site-types` instead.